smac/intensification/successive_halving.py

```diff
@@ -291,7 +294,7 @@ def eval_challenger(self,
         #   - this check is required because there is no incumbent performance
         #     for the first ever 'intensify' run (from initial design)
         if incumbent:
-            inc_runs = run_history.get_runs_for_config(incumbent)
+            inc_runs = run_history.get_runs_for_config(incumbent, only_max_observed_budget=True)
```

Comment:
Check why line 304 is different from line 421

Reply:
I simplified the code to remove the confusion here.